### ISSUE TITLE
[ssl] in-epoch Validation

### ISSUE THREAD

**Issue by sbatchelder** [WHOIGit/image-classifier-dojo#10]
Self Supervised Learning requires a lot more data instances than Supervised Learning. This leads to epochs that span multiple days which feels like a very long time during which only loss metrics are collected. 

It's apocryphal, but what if we could perform validation with labeled data before the end of an unlabeled-training epoch? So we can track the performance of a long-running SSL training while still training on a billion+ total ROIs. 

The dataloader would have to be able to keep track of these "sub-epochs" so as to not reload previous data, perhaps something similar to how workers are handled for an iterable dataloader. Additionally, when a "true" full epoch is completed, the data would have to be re-included and re-shuffled.

**Comment by sbatchelder** [comment]
Turns out this is very easy to do with `pl.Trainer(val_check_interval=...)`  argument. It can accept a step number (counts training batches) or a float representing a fractions of an epoch. Super handy
